### DISCUSSION TITLE
Indented epigraph

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4927] [Q&A] [answered]
Currently, to place an epigraph, I use a 2-column `grid`, the first cell of which is empty:

```
#grid(
  columns: (65%, auto),
  grid.cell([]),
  grid.cell([
    Roses are red, \
    Violets are blue lorem ipsum dolor \
    etc. \
    — John Doe
  ]),
)
```

But is it really the correct way? Maybe there is a more correct, simpler approach?

Note that I don't mean a right-aligned epigraph. That is, I know the following works, but this is not what I need:

```
#align(right)[
  Roses are red, \
  Violets are blue lorem ipsum dolor \
  etc. \
  — John Doe
]
```

**Comment by EpicEricEE** [comment, marked as answer]
I can think of two options:

1. The [`pad`](https://typst.app/docs/reference/layout/pad/) function (when you know how much on the right it should be):
  
   ```typ
   #pad(left: 65%)[
     Roses are red, \
     Violets are blue lorem ipsum dolor \
     etc. \
     — John Doe
   ]
   ```
2. Right-aligning a left aligned block:
   
   ```typ
   #align(right, block(align(left)[
     Roses are red, \
     Violets are blue lorem ipsum dolor \
     etc. \
     — John Doe
   ]))
   ```
